core/include/crypto/aes-gcm.h

```diff
@@ -22,13 +22,12 @@ struct aes_gcm_ctx {
 	uint8_t buf_hash[TEE_AES_BLOCK_SIZE];
 	uint8_t buf_cryp[TEE_AES_BLOCK_SIZE];
 
-	unsigned tag_len;
-	unsigned aad_bytes;
-	unsigned payload_bytes;
-
 	symmetric_key skey;
 
-	unsigned buf_pos;
+	unsigned int tag_len;
+	unsigned int aad_bytes;
+	unsigned int payload_bytes;
+	unsigned int buf_pos;
```

Comment:
`size_t` ?

Reply:
Using a 64-bit value (on 64-bit platforms) seems a bit wasteful since we're only storing small numbers. It could be argued that `aad_bytes` or `payload_bytes` should be 64-bit values, but I doubt we'll need it in practice. I'll change them if you think it's important.